lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Well, it's something like this: https://gist.github.com/cmrd-senya/a8263ec0394f7b24ce3b989ef296b274

It can look more complicated since it now contains more classes, but I think that it's actually better since different behaviors are split into different classes now. This way it is explicit which methods are used for submitting a job to sidekiq and which methods are used for entity dispatching. And if we change these classes in future it is less likely to get into hardly maintainable code by mixing different responsibilities. So I think it is better in terms of maintainability.

I'm fine if you decide to reject this code though. I think can live with how we have it now. But I think that this change slightly improves the implementation.